ocaml-lsp-server/src/compl.ml

```diff
@@ -26,8 +26,16 @@ let completion_kind kind : CompletionItemKind.t option =
   | `Constructor -> Some Constructor
   | `Type -> Some TypeParameter
 
-(** @see <https://ocaml.org/manual/lex.html> reference *)
-let prefix_of_position ~short_path source position =
+(* I should just rewrite all of this so that it uses a nice for loop. This
+   current soluction is a nice try but overall crap we need to be able to look
+   ahead and behind
+
+   Split it into name and infix name is obvious infix can be either a dot, a
+   label or an I could possibly do a regex based parser.
+
+   Name regex: ((\w)|\w.)*$ *)
+
+let prefix_of_position_old ~short_path source position =
```

Comment:
Let's remove this function unless it's used elsewhere. We shouldn't keep it just for benchmarks.